builder/builder.go

```diff
@@ -366,7 +366,7 @@ func (b *Builder) convertAndBuild(ctx context.Context, target domain.Target, opt
 		if hasRunPush {
 			err = b.s.buildMainMulti(ctx, bf, onImage, onArtifact, onFinalArtifact, "--push")
 			if err != nil {
-				return nil, errors.Wrapf(err, "build push")
+				return nil, errors.Wrapf(NewBuildError(err, b.s.vertexFailureOutput), "build push")
```

Comment:
Could `buildMainMulti` return this error itself? Maybe a defer could do a wrapping if err != nil.

Reply:
Good point, I moved it there instead.